class.jetpack-cli.php

```diff
@@ -1476,6 +1476,156 @@ private function partner_provision_error( $error ) {
+				exit( 1 );
+		}
+	}
+
+	/**
+	 * Creates the essential files in Jetpack to build a Gutenberg block.
+	 *
+	 * @param array $args       Positional parameters. Only one is used, that corresponds to the block title.
+	 * @param array $assoc_args Associative parameters defined in the scaffold() method.
+	 */
+	public function block( $args, $assoc_args ) {
+		if ( isset( $args[1] ) ) {
+			$title = $args[1];
+		} else {
+			WP_CLI::error( esc_html__( 'The title parameter is required.', 'jetpack' ) . ' 👻' );
+			exit( 1 );
+		}
+
+		$slug = isset( $assoc_args['slug'] )
+			? $assoc_args['slug']
+			: sanitize_title( $title );
+
+		if ( ! preg_match( '/^[a-z][a-z0-9\-]*$/', $slug ) ) {
+			WP_CLI::error( esc_html__( 'Invalid block slug. They can contain only lowercase alphanumeric characters or dashes, and start with a letter', 'jetpack' ) . ' 👻' );
+		}
+
+		global $wp_filesystem;
+		if ( ! WP_Filesystem() ) {
+			WP_CLI::error( esc_html__( "Can't write files", 'jetpack' ) . ' 😱' );
+		}
+
+		$path = JETPACK__PLUGIN_DIR . "extensions/blocks/$slug";
+
+		if ( $wp_filesystem->exists( $path ) && $wp_filesystem->is_dir( $path ) ) {
+			WP_CLI::error( sprintf( esc_html__( 'Name conflicts with the existing block %s', 'jetpack' ), $path ) . ' ⛔️' );
+			exit( 1 );
+		}
+
+		$wp_filesystem->mkdir( $path );
+
+		$hasKeywords = isset( $assoc_args['keywords'] );
+
+		$files = array(
+			"$path/$slug.php" => $this->render_block_file( 'block-register-php', array(
+				'slug' => $slug,
+				'title' => $title,
+				'underscoredSlug' => str_replace( '-', '_', $slug ),
+			) ),
+			"$path/index.js" => $this->render_block_file( 'block-index-js', array(
+				'slug' => $slug,
+				'title' => $title,
+				'description' => isset( $assoc_args['description'] )
+					? $assoc_args['description']
+					: $title,
+				'keywords' => $hasKeywords
+					? array_map( function( $keyword ) {
+						// Construction necessary for Mustache lists
+						return array( 'keyword' => trim( $keyword ) );
+					}, explode( ',', $assoc_args['keywords'] ) )
+					: '',
+				'hasKeywords' => $hasKeywords
+			) ),
+			"$path/editor.js" => $this->render_block_file( 'block-editor-js' ),
+			"$path/editor.scss" => $this->render_block_file( 'block-editor-scss', array(
+				'slug' => $slug,
+				'title' => $title,
+			) ),
+			"$path/edit.js" => $this->render_block_file( 'block-edit-js', array(
+				'title' => $title,
+				'className' => str_replace( ' ', '', ucwords( str_replace( '-', ' ', $slug ) ) ),
+			) )
+		);
+
+		$files_written = array();
+
+		foreach ( $files as $filename => $contents ) {
+			if ( $wp_filesystem->put_contents( $filename, $contents ) ) {
+				$files_written[] = $filename;
+			} else {
+				WP_CLI::error( sprintf( esc_html__( 'Error creating %s', 'jetpack' ), $filename ) );
+			}
+		}
+
+		if ( empty( $files_written ) ) {
+			WP_CLI::log( esc_html__( 'No files were created' ) );
```

Comment:
Do we want to add a textdomain here?

```suggestion
			WP_CLI::log( esc_html__( 'No files were created', 'jetpack' ) );
```

Reply:
👍